docs/modules/selfserve.html

```diff
@@ -0,0 +1,338 @@
+						</div>
+					</div>
+					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
+				</div>
+			</div>
+		</div>
+	</div>
+	<div class="tsd-page-title">
+		<div class="container">
+			<ul class="tsd-breadcrumb">
+				<li>
+					<a href="../modules.html">cosmos-explorer</a>
+				</li>
+				<li>
+					<a href="selfserve.html">SelfServe</a>
+				</li>
+			</ul>
+			<h1>Module SelfServe</h1>
+		</div>
+	</div>
+</header>
+<div class="container container-main">
+	<div class="row">
+		<div class="col-8 col-content">
+			<section class="tsd-panel tsd-comment">
+				<div class="tsd-comment tsd-typography">
+					<div class="lead">
+						<a href="#self-serve-model" id="self-serve-model" style="color: inherit; text-decoration: none;">
+							<h1>Self Serve Model</h1>
+						</a>
+						<p>The Self Serve Model allows you to write classes that auto generate UI components for your feature. The idea is to allow developers, who aren&#39;t very familiar with writing UI, to develop and own UI components for their features by just writing simpler TypeScript classes. </p>
+						<p>What this means for the feature team </p>
+						<ul>
+							<li>Can concentrate just on the logic behind showing, hiding and disabling UI components </li>
+							<li>Need not worry about specifics of the UI language</li>
+							<li>Can own the REST API calls made as part of the feature, which may change in the future</li>
+							<li>Quicker turn around time for feature bugs, which they own and have deeper knowledge of</li>
+						</ul>
+						<p>What this means for the UI team</p>
+						<ul>
+							<li>Lesser feature development time, since they only own the framework and not the feature itself</li>
+						</ul>
+						<p>Each team owns what they know best, a win-win!</p>
+						<a href="#getting-started" id="getting-started" style="color: inherit; text-decoration: none;">
+							<h2>Getting Started</h2>
+						</a>
+						<p>Clone the cosmos-explorer repo and run</p>
+						<ul>
+							<li><code>npm install</code></li>
+							<li><code>npm run build</code></li>
+						</ul>
+						<a href="#developing" id="developing" style="color: inherit; text-decoration: none;">
+							<h2>Developing</h2>
+						</a>
+						<a href="#watch-mode" id="watch-mode" style="color: inherit; text-decoration: none;">
+							<h3>Watch mode</h3>
+						</a>
+						<p>Run <code>npm start</code> to start the development server and automatically rebuild on changes</p>
+						<a href="#local-development" id="local-development" style="color: inherit; text-decoration: none;">
+							<h3>Local Development</h3>
+						</a>
+						<p>Update <a href="../enums/selfserve_selfserveutils.selfservetype.html">SelfServeType</a> to add your feature.</p>
+						<p>Add the feature flag <code>feature.dataExplorerSource=https://localhost:1234/selfServe.html?selfServeType%3D&lt;SELF_SERVE_TYPE&gt;</code> to open up the your feature&#39;s UI in the data explorer blade of the portal.</p>
```

Comment:
>e%3D [](http://example.com/codeflow?start=113&length=4)

you dont need encoded url here

Reply:
So I tried adding the feature flag without encoding it - 
https://ms.portal.azure.com/?feature.dataExplorerSource=https://localhost:1234/selfServe.html?selfServeType=example
The "example" selfServeType is not picked up for some reason. encoding the '=' makes i work through. It might be a change we need to make on the portal side of things, to decode the url for dataExplorerSource correctly